src/main/java/seedu/exercise/logic/commands/events/AddExerciseEvent.java

```diff
@@ -18,10 +19,10 @@
     /**
      * Creates an AddExerciseEvent to store the particular event of an exercise being added to the exercise book.
      *
-     * @param exercise the exercise that has been added in this instance of AddExerciseEvent.
+     * @param eventPayload a wrapper class that stores the essential information for undo and redo
      */
-    AddExerciseEvent(Exercise exercise) {
-        this.exercise = exercise;
+    AddExerciseEvent(EventPayload<Exercise> eventPayload) {
+        this.exercise = eventPayload.get(AddExerciseCommand.KEY_EXERCISE_TO_ADD);
```

Comment:
Since the `KEY` for the `EventPayload` objects are used to retrieve the values here, i feel that the `KEY` should be a constant variable in the `Event` classes. Then the `AddCommand` can simply select which key they wish to add to the `Event` classes and the class have full control over the `KEY` that are being used. Nothing huge really. Just where to place the constants

Reply:
Amended